src/legacy/ui/public/url/url.js

```diff
@@ -188,28 +208,30 @@ export function KbnUrlProvider($injector, $location, $rootScope, $parse, Private
 
   self._changeLocation = function (type, url, paramObj, replace, appState) {
     const prev = {
-      path: $location.path(),
-      search: $location.search()
+      path: location.path(),
+      search: location.search()
     };
 
     url = self.eval(url, paramObj);
-    $location[type](url);
-    if (replace) $location.replace();
+    location[type](url, replace);
 
     if (appState) {
-      $location.search(appState.getQueryParamName(), appState.toQueryParam());
+      location.search(appState.getQueryParamName(), appState.toQueryParam());
     }
 
     const next = {
-      path: $location.path(),
-      search: $location.search()
+      path: location.path(),
+      search: location.search()
     };
 
-    if ($injector.has('$route')) {
+    const $injector = chrome.dangerouslyGetActiveInjector();
```

Comment:
we use $injector to limit angular dependency just to this part (access to angular router). we can remove it once we have a new platform router

Reply:
`chrome.dangerouslyGetActiveInjector()` is async. I'm curious how that might effect code that uses this.